decorator.rb

```diff
@@ -0,0 +1,26 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    original_name = @nameable.correct_name
+    original_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    original_name = @nameable.correct_name
+    original_name.length <= 10 ? original_name : original_name[0, 10]
+  end
+end
```

Comment:
- [ OPTIONAL ] Hello there! 📂 It's generally regarded as a best practice to have each class reside in its own dedicated file. 📦 For enhanced organization and maintainability, it would be a good idea to create separate files for the CapitalizeDecorator and TrimmerDecorator classes. 📝 This approach not only aligns with coding conventions but also makes your codebase cleaner and more manageable. 🧹 What are your thoughts on this suggestion? 🤔

Reply:
Thanks so much for the suggestions . I will implement the changes